src/doc/rustdoc/src/lints.md

```diff
@@ -285,3 +285,40 @@ warning: unclosed HTML tag `h1`
 
 warning: 2 warnings emitted
 ```
+
+## automatic_links
+
+This lint is **nightly-only** and **warns by default**. It detects links which
+could use the "automatic" link syntax. For example:
+
+```rust
+/// http://hello.rs
+/// [http://a.com](http://a.com)
+/// [http://b.com]
+///
+/// [http://b.com]: http://b.com
```

Comment:
You need to use domains like `example.com` in documentation not just random domains. See https://en.wikipedia.org/wiki/Example.com.